src/VisualStudio/VisualStudioDiagnosticsToolWindow/Loggers/OutputWindowLogger.cs

```diff
@@ -41,97 +56,27 @@ public bool IsEnabled(FunctionId functionId)
 
         public void Log(FunctionId functionId, LogMessage logMessage)
         {
-            OutputPane.WriteLine(string.Format("[{0}] {1} - {2}", Thread.CurrentThread.ManagedThreadId, functionId.ToString(), logMessage.GetMessage()));
+            WriteLine(string.Format("[{0}] {1} - {2}", Thread.CurrentThread.ManagedThreadId, functionId.ToString(), logMessage.GetMessage()));
         }
 
         public void LogBlockStart(FunctionId functionId, LogMessage logMessage, int uniquePairId, CancellationToken cancellationToken)
         {
-            OutputPane.WriteLine(string.Format("[{0}] Start({1}) : {2} - {3}", Thread.CurrentThread.ManagedThreadId, uniquePairId, functionId.ToString(), logMessage.GetMessage()));
+            WriteLine(string.Format("[{0}] Start({1}) : {2} - {3}", Thread.CurrentThread.ManagedThreadId, uniquePairId, functionId.ToString(), logMessage.GetMessage()));
         }
 
         public void LogBlockEnd(FunctionId functionId, LogMessage logMessage, int uniquePairId, int delta, CancellationToken cancellationToken)
         {
             var functionString = functionId.ToString() + (cancellationToken.IsCancellationRequested ? " Canceled" : string.Empty);
-            OutputPane.WriteLine(string.Format("[{0}] End({1}) : [{2}ms] {3}", Thread.CurrentThread.ManagedThreadId, uniquePairId, delta, functionString));
+            WriteLine(string.Format("[{0}] End({1}) : [{2}ms] {3}", Thread.CurrentThread.ManagedThreadId, uniquePairId, delta, functionString));
         }
 
-        private class OutputPane
+        private void WriteLine(string value)
         {
-            private static readonly Guid s_outputPaneGuid = new Guid("BBAFF416-4AF5-41F2-9F93-91F283E43C3B");
-
-            public static readonly OutputPane s_instance = new OutputPane();
-
-            private readonly IServiceProvider _serviceProvider;
-            private readonly IThreadingContext _threadingContext;
-
-            public static void WriteLine(string value)
-            {
-                s_instance.WriteLineInternal(value);
-            }
-
-            public OutputPane()
-            {
-                _serviceProvider = ServiceProvider.GlobalProvider;
-
-                var componentModel = (IComponentModel)_serviceProvider.GetService(typeof(SComponentModel));
-                _threadingContext = componentModel.GetService<IThreadingContext>();
-            }
-
-            private IVsOutputWindowPane _doNotAccessDirectlyOutputPane;
-
-            private void WriteLineInternal(string value)
-            {
-                var pane = GetPane();
-                if (pane == null)
-                {
-                    return;
-                }
-
-                pane.OutputStringThreadSafe(value + Environment.NewLine);
-            }
-
-            private IVsOutputWindowPane GetPane()
+            _threadingContext.JoinableTaskFactory.RunAsync(async () =>
```

Comment:
The caller is synchronous so this is expected to block on completion of the underlying write.

```suggestion
            _threadingContext.JoinableTaskFactory.Run(async () =>
```

Reply:
This could now be a request over the wire to a remote server.  I'm not sure that we want it to block on network requests.  

I looked at propagating the async up, but then we need to modify ILogger and all the implementations to be async.  I wasn't sure if that was something we should be doing for this one.